prac_07/project_management.py

```diff
@@ -0,0 +1,199 @@
+        print(f"{i} {new_project}")
+    project_choice = validate_detail("Project choice: ", int, MINIMUM_PROJECT_CHOICE, len(data) - 1)
+    print(data[project_choice])
+    update_percentage(data, project_choice)
+    update_priority(data, project_choice)
+
+
+def update_priority(data, project_choice):
+    """Update priority."""
+    is_valid_input = False
+    while not is_valid_input:
+        try:
+            new_priority = input("New Priority: ")
+            if new_priority != "":
+                if int(new_priority) < MINIMUM_PRIORITY or int(new_priority) > MAXIMUM_PRIORITY:
+                    print("Error priority.")
+                else:
+                    data[project_choice].priority = int(new_priority)
+                    is_valid_input = True
+            else:
+                is_valid_input = True
+        except ValueError:
+            print("Priority should be a number.")
+
+
+def update_percentage(data, project_choice):
+    """Update percentage."""
+    is_valid_input = False
+    while not is_valid_input:
+        try:
+            new_percentage = input("New Percentage: ")
+            if new_percentage != "":
+                if int(new_percentage) < MINIMUM_PERCENTAGE or int(new_percentage) > MAXIMUM_PERCENTAGE:
+                    print("Error percentage.")
+                else:
+                    data[project_choice].completion_percentage = int(new_percentage)
+                    is_valid_input = True
+            else:
+                is_valid_input = True
+        except ValueError:
+            print("Percentage should be a number.")
+
+
+def save_data(data, filename):
+    """Save data into the txt file."""
+    saving_message = input("Would you like to save to projects.txt? ")
+    if saving_message == "Yes":
+        with open(filename, 'w') as out_file:
+            print("Name\tStart Date\tPriority\tCost Estimate\tCompletion Percentage", file=out_file)
+            for project in data:
+                print(f"{project.name}\t{project.start_date}\t{project.priority}\t{project.cost_estimate}\t"
+                      f"{project.completion_percentage}", file=out_file)
+
+
+def add_new_project(data):
+    """Add new project."""
+    print("Let's add a new project")
+    name = input("Name: ").title()
+    start_date = input("Start date (dd/mm/yy): ")
+    start_date = datetime.datetime.strptime(start_date, "%d/%m/%Y").date()
+    priority = validate_detail("Priority: ", int, MINIMUM_PRIORITY, MAXIMUM_PRIORITY)
+    cost_estimate = validate_detail("Cost estimate: $", float, MINIMUM_COST_ESTIMATE, MAXIMUM_COST_ESTIMATE)
+    completion_percentage = validate_detail("Percent complete: ", int, MINIMUM_PERCENTAGE, MAXIMUM_PERCENTAGE)
+    new_project = Project(name, start_date.strftime("%d/%m/%Y"), priority, cost_estimate, completion_percentage)
+    data.append(new_project)
+
+
+def validate_detail(message, detail_type, minimum_limit, maximum_limit):
+    """Validate element."""
```

Comment:
The role of this function is very ambiguous. "Validate element" doesn't give me a good idea of what it is actually doing.